jina/serve/runtimes/gateway/__init__.py

```diff
@@ -125,14 +125,14 @@ async def _wait_for_cancel(self):
 
     async def async_teardown(self):
         """Shutdown the server."""
-        await self.cancel_warmup_task()
+        self.cancel_warmup_task()
```

Comment:
I think this will cause some other issues because the task is relying on the gRPC connection pools. If the `gateway.streamer.close()` will execute before the shutdown then it will again lead to a seg fault on the warmup task.

Reply:
but those task are safe to cancel